src/validation/variable.rs

```diff
@@ -101,16 +102,47 @@ fn validate_vla(validator: &mut Validator, pou: Option<&Pou>, block: &VariableBl
     }
 }
 
+fn validate_array_ranges<T>(validator: &mut Validator, variable: &Variable, context: &ValidationContext<T>)
+where
+    T: AnnotationMap,
+{
+    let ty_name = variable.data_type_declaration.get_name().unwrap_or_default();
+    let ty_info = context.index.get_effective_type_or_void_by_name(ty_name).get_type_information();
+
+    if ty_info.is_array() {
+        let mut types = vec![];
+        ty_info.get_inner_types(&mut types, context.index);
```

Comment:
Not the best API design I must admit but I think it's fine as it is effectively only used here right now? An alternative approach would have been to not pass a vector by the caller but rather the callee return a vector by itself.